Sources/EmbraceIO/EmbraceIO+Metadata.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  Copyright © 2025 Embrace Mobile, Inc. All rights reserved.
+//
+
+/// User QoL accessors
+extension EmbraceIO {
+
+    /// Set a 'name' for the current user.
+    /// Will be set permanently until explicitly unset via the `clearUserProperties()` method.
+    /// - Note: No validation is done on the identifier. Be sure it matches or can be mapped to a record in your system
+    public var userName: String? {
+        get {
+            Embrace.client?.metadata.userName
+        }
+        set {
+            Embrace.client?.metadata.userName = newValue
+        }
+    }
+
+    /// Set a 'email' for the current user.
+    /// Will be set permanently until explicitly unset via the `clearUserProperties()` method.
+    /// - Note: No validation is done on the username. Be sure it matches or can be mapped to a record in your system
+    public var userEmail: String? {
+        get {
+            Embrace.client?.metadata.userEmail
+        }
+        set {
+            Embrace.client?.metadata.userEmail = newValue
+        }
+    }
+
+    /// Set an 'identifier' for the current user.
+    /// Will be set permanently  until explicitly unset via the `clearUserProperties()` method.
+    /// - Note: No validation is done on the email address. Be sure it matches or can be mapped to a record in your system
+    public var userIdentifier: String? {
+        get {
+            Embrace.client?.metadata.userIdentifier
+        }
+        set {
+            Embrace.client?.metadata.userIdentifier = newValue
+        }
+    }
+
+    /// Clear all user properties.
+    /// This will clear all user properties set via the `userName`, `userEmail` and `userIdentifier` properties.
+    public func clearUserProperties() {
+        Embrace.client?.metadata.clearUserProperties()
+    }
+}
+
+/// Session properties
+extension EmbraceIO {
+
+    /// Adds or removes a property with the given key, value and lifespan.
+    /// If there are 2 properties with the same key but different lifespans, the one with a shorter lifespan will be used.
+    /// - Parameters:
+    ///   - key: The key of the property to add. Can not be longer than 128 characters.
+    ///   - value: The value of the property to add. Will be truncated if its longer than 1024 characters.
+    ///   - lifespan: The lifespan of the property to add.
+    /// - Throws: `MetadataError.invalidKey` if the key is longer than 128 characters.
+    /// - Throws: `MetadataError.invalidSession` if a property with a `.session` lifespan is added when there's no active session.
+    public func setProperty(key: String, value: String?, lifespan: MetadataLifespan) throws {
```

Comment:
```suggestion
    public func setProperty(for key: String, value: String?, lifespan: MetadataLifespan = .process) throws {
```

Reply:
It would be great to have a version of this that takes in a [String: String]. It's not rare to have to set more than one key/value at a time and it becomes a burden on perf and efficiency to loop over them when we could internally make one call.